File: benchmark/results/java-invocation-inline/run-002/b-baseline/output/com/metaengine/demo/billing/services/InvoiceRepository.java
package com.metaengine.demo.billing.services;

import java.util.List;

import com.metaengine.demo.billing.aggregates.Invoice;

/** InvoiceRepository for the billing domain. */
public class InvoiceRepository {

    /** Persist a new Invoice from partial input. */
    public Invoice create(Invoice input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find an Invoice by id. Returns null if not found. */
    public Invoice findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List Invoices up to the given limit. */
    public List<Invoice> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete the Invoice with the given id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-multilang/run-003/b-baseline/output/metaengine_demo/billing/services/invoice_repository.py
"""InvoiceRepository for the billing domain."""
from typing import Optional

from metaengine_demo.billing.aggregates.invoice import Invoice


class InvoiceRepository:
    """InvoiceRepository persistence service."""

    def create(self, input: Optional[Invoice]) -> Invoice:
        """Persist a new invoice."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Invoice]:
        """Find an invoice by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Invoice]:
        """List invoices up to a limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete an invoice by id."""
        raise NotImplementedError("not implemented")
